tests/Unit/Handler/AddProductBundleToCartHandlerTest.php

```diff
@@ -0,0 +1,348 @@
+namespace Tests\BitBag\SyliusProductBundlePlugin\Unit\Handler;
+
+use BitBag\SyliusProductBundlePlugin\Command\AddProductBundleToCartCommand;
+use BitBag\SyliusProductBundlePlugin\Entity\OrderItemInterface;
+use BitBag\SyliusProductBundlePlugin\Entity\ProductBundle;
+use BitBag\SyliusProductBundlePlugin\Entity\ProductBundleInterface;
+use BitBag\SyliusProductBundlePlugin\Entity\ProductBundleItem;
+use BitBag\SyliusProductBundlePlugin\Entity\ProductBundleItemInterface;
+use BitBag\SyliusProductBundlePlugin\Entity\ProductBundleOrderItem;
+use BitBag\SyliusProductBundlePlugin\Entity\ProductBundleOrderItemInterface;
+use BitBag\SyliusProductBundlePlugin\Factory\ProductBundleOrderItemFactoryInterface;
+use BitBag\SyliusProductBundlePlugin\Handler\AddProductBundleToCartHandler;
+use Doctrine\ORM\EntityManagerInterface;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Sylius\Component\Core\Model\Order;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\Product;
+use Sylius\Component\Core\Model\ProductVariant;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Order\Modifier\OrderItemQuantityModifierInterface;
+use Sylius\Component\Order\Modifier\OrderModifierInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Tests\BitBag\SyliusProductBundlePlugin\Entity\OrderItem;
+use Webmozart\Assert\InvalidArgumentException;
+
+final class AddProductBundleToCartHandlerTest extends TestCase
+{
+    /** @var mixed|MockObject|OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var mixed|MockObject|RepositoryInterface */
+    private $productBundleRepository;
+
+    /** @var mixed|MockObject|FactoryInterface */
+    private $orderItemFactory;
+
+    /** @var ProductBundleOrderItemFactoryInterface|mixed|MockObject */
+    private $productBundleOrderItemFactory;
+
+    /** @var mixed|MockObject|OrderModifierInterface */
+    private $orderModifier;
+
+    /** @var mixed|MockObject|OrderItemQuantityModifierInterface */
+    private $orderItemQuantityModifier;
+
+    /** @var EntityManagerInterface|mixed|MockObject */
+    private $orderManager;
+
+    protected function setUp(): void
+    {
+        $this->orderRepository = $this->createMock(OrderRepositoryInterface::class);
+        $this->productBundleRepository = $this->createMock(RepositoryInterface::class);
+        $this->orderItemFactory = $this->createMock(FactoryInterface::class);
+        $this->productBundleOrderItemFactory = $this->createMock(ProductBundleOrderItemFactoryInterface::class);
+        $this->orderModifier = $this->createMock(OrderModifierInterface::class);
+        $this->orderItemQuantityModifier = $this->createMock(OrderItemQuantityModifierInterface::class);
+        $this->orderManager = $this->createMock(EntityManagerInterface::class);
```

Comment:
I'm wonder if number of dependencies is not too high. Can this handler be broken into a few, easier testable classes? I'm afraid that current coverage is pretty low due to the amount of dependencies.